samtranslator/model/appsync.py

```diff
@@ -7,11 +7,44 @@
 from samtranslator.model.types import IS_DICT, IS_STR, is_type, list_of
 from samtranslator.utils.types import Intrinsicable
 
+# Data source constants can be found here under "Type" property:
+# https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appsync-datasource.html
+SUPPORTED_DATASOURCES = {"AMAZON_DYNAMODB"}
 
+
+# ====== common types =======
+class DeltaSync(TypedDict):
+    BaseTableTTL: str
+    DeltaSyncTableName: str
+    DeltaSyncTableTTL: str
+
+
+# ====== serverless types ======
 class Auth(TypedDict, total=False):
     Type: str
 
 
+class DynamoDBDataSource(TypedDict, total=False):
+    Region: str
+    TableName: str
+    UseCallerCredentials: bool
+    Versioned: bool
+    DeltaSync: DeltaSync
+
+
+class DataSourceConfig(TypedDict, total=False):
```

Comment:
Curious why is this class `total=False`? My understanding is that `total=False` doesn't require all keys present when instantiating a class.

Reply:
Currently that is true but we will be adding support for more data sources. I'm going to leave it for now since it will change in the near future and it's not impacting anything.